GLAA.Web/FormLogic/LicenceApplicationFormDefinition.cs

```diff
@@ -38,11 +52,115 @@ public bool CanViewNextModel<TParent>(FormSection section, int id, TParent paren
             return model == null || model.CanView(parent);
         }
 
+        public bool CanViewPage<TParent>(FormSection section, string actionName, TParent parent)
+        {
+            var page = GetPageDefinition(section, actionName);
+
+            if (page.OverrideViewCondition)
+            {
+                return true;
+            }
+
+            var model = page.GetViewModelExpressionForPage(parent) as ICanView<TParent>;
+
+            return model == null || model.CanView(parent);
+        }
+
         public int GetSectionLength(FormSection section)
         {
-            return fieldConfiguration.Fields[section].Length;
+            return fieldConfiguration[section].Length;
         }
 
+        public FormPageDefinition GetLastPage(FormSection section)
+        {
+            return fieldConfiguration[section].Last();
+        }
+
+        public bool IsNextPageLastPage(FormSection section, string actionName)
+        {
+            return fieldConfiguration[section][GetSectionLength(section) - 2].MatchesName(actionName);
+        }
+
+        public FormPageDefinition GetNextPage(FormSection section, string actionName)
+        {
+            // Get the index of the page that matches this action name
+            for (var i = 0; i < GetSectionLength(section); i++)
+            {
+                if (fieldConfiguration[section][i].MatchesName(actionName))
+                {
+                    // return the next page in the sequence
+                    return fieldConfiguration[section][i + 1];
+                }
+            }
+
+            // if in doubt, return the final (summary) screen
+            return GetLastPage(section);
+        }
+
+        public FormPageDefinition GetPreviousPage(FormSection section, string actionName)
+        {
+            // Counting back from the end, find the page that matches this action name
+            for (var i = GetSectionLength(section) - 1; i > 0; i--)
```

Comment:
Could this be >= and avoid the -1?

Reply:
Not in this case, we're counting backwards from the last item in the list so we're going to have to use it somewhere ☹️